metadata-ws/src/test/java/uk/ac/ebi/ampt2d/metadata/cucumber/CommonSteps.java

```diff
@@ -108,29 +118,47 @@ public void cleanStates() {
 
     @When("^I request GET ([\\S]*)$")
     public void performGetOnResourceUri(String resourceUri) throws Exception {
+        CommonStates.setResultActions(mockMvc.perform(get(resourceUri)
+                .with(CommonStates.getRequestPostProcessor()))
+        );
+    }
+
+    @When("^I request anonymous GET ([\\S]*)$")
```

Comment:
Currently we have three possible ways to make a test request:

- Anonymous (`I request anonymous GET`)
- Via an unprivileged test user (`I set authorization with testuser having default role`)
- Via a privilege service operator (`I set authorization with testoperator having SERVICE_OPERATOR role`)

Given that we decided to only have two states (either anonymous, or service operator), I think one of the above modes should be removed. If possible, I think it would be ideal to have two operations:
- `I set authorization with testoperator having SERVICE_OPERATOR role` (switches the request post processor to SERVICE_OPERATOR role)
- `I reset authorization` (removes/resets request post processor)

Reply:
The reason why i retained `I set authorization with testuser having default role` is there are some tests where write operation is tried with this role and failed. Just to make sure user with only SERVICE_OPERATOR role allowed for write operation and others are not. 
It can be changed aswell.

Reply:
My point is that currently the tests simulate an unprivileged agent by using a `testuser` _with a default role;_ while in reality unprivileged agents will be not authenticated at all. However I don't think this makes a difference for Spring Security, so I won't press for these edits. Considering this resolved.